### PR TITLE
Add GitLab CI credentials helper

### PR DESCRIPTION
**Pull request by ePirat**
**Description**

Adds a GitLab CI credentials helper that fetches the credentials
for a GitLab Docker registry from the CI environment if available.

**Submitter Checklist**

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [unit tests](../DEVELOPMENT.md#creating-a-pr)
- [ ] Adds integration tests if needed.

**Reviewer Notes**

- [ ] The code flow looks good. 
- [ ] Unit tests and or integration tests added.

**Release Notes**

```
- kaniko now automatically uses GitLab CI Docker registry credentials from the env variables, if available
```

### CODE REVIEW

**Review by imjasonh — COMMENTED** [review]
Thanks for this! Always great to see another cred helper implementation 👍

File: deploy/Dockerfile_debug [hunk 1]
```diff
@@ -31,6 +31,9 @@ RUN go install github.com/awslabs/amazon-ecr-credential-helper/ecr-login/cli/doc
 # Get ACR docker env credential helper
 RUN go install github.com/chrismellard/docker-credential-acr-env@09e2b5a8ac86c3ec347b2473e42b34367d8fa419
 
+# Get GitLab CI env credential helper
+RUN go install github.com/ePirat/docker-credential-gitlabci@10832279bfb8014babd85f3acef86600c59e09e4 # v1.0.0
```
[the remark this answers is not in the document]
Comment (ePirat) [reply deploy/Dockerfile_debug:35]: Sounds good to me, indeed it is not necessary, was just useful during development to query it easily to debug things.

**Review by imjasonh — APPROVED** [review]

**Comment by ePirat** [comment]
Thanks for the review and merging. I was wondering, how long will it take for the new version to be available at gcr.io? Will it require a formal release or when are these images updated?

**Comment by imjasonh** [comment]
Commit-tagged images are available. The `debug` variant was just pushed 6 minutes ago, to `gcr.io/kaniko-project/executor:f930b75b8b2c310b3d7fc4e1d8ddb30859cee9aa-debug` -- the rest should be showing up soon (watch progress [here](https://github.com/GoogleContainerTools/kaniko/actions/runs/2150069063))

**Comment by ePirat** [comment]
Thanks, and I guess the `gcr.io/kaniko-project/executor:debug` will be updated once the next release is tagged, right?

**Comment by imjasonh** [comment]
> Thanks, and I guess the `gcr.io/kaniko-project/executor:debug` will be updated once the next release is tagged, right?

Yep, exactly.

**Pull request merged**